distributed/scheduler.py

```diff
@@ -2059,6 +2059,93 @@ def transition_no_worker_processing(self, key, stimulus_id):
                 pdb.set_trace()
             raise
 
+    def transition_no_worker_erred(
+        self,
+        key,
+        stimulus_id,
+        cause: str | None = None,
+        exception=None,
+        traceback=None,
+        exception_text: str | None = None,
+        traceback_text: str | None = None,
+        **kwargs,
+    ):
```

Comment:
I suggest to add a comment mentioning shuffling as a use case for this.
I would also mention that this could/should be used to enable the feature we've been talking about about failing a task if the worker restrictions workers left assuming they cannot reconnect any longer

Reply:
Done